### PR TITLE
Factor out common code to make a EntryRef vector snapshot.

### PR DESCRIPTION
**Pull request by toregge**
@geirst  : please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/save_utils.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#include <vespa/vespalib/datastore/atomic_entry_ref.h>
+#include <vespa/vespalib/datastore/entryref.h>
+#include <vespa/vespalib/stllike/allocator.h>
+#include <vespa/vespalib/util/rcuvector.h>
+#include <vector>
+
+namespace search::attribute {
+
+using EntryRefVector = std::vector<vespalib::datastore::EntryRef, vespalib::allocator_large<vespalib::datastore::EntryRef>>;
+
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/save_utils.h:14]: Please add function description.

**Review by geirst — DISMISSED** [review]
👍

File: searchlib/src/vespa/searchlib/attribute/save_utils.h (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/save_utils.h:14]: Added.

File: searchlib/src/vespa/searchlib/attribute/save_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+#include "save_utils.h"
+#include <cassert>
+
+namespace search::attribute {
+
+EntryRefVector
+make_entry_ref_vector_snapshot(const vespalib::RcuVectorBase<vespalib::datastore::AtomicEntryRef>& ref_vector, uint32_t size)
+{
+    assert(size <= ref_vector.get_size());      // Called from writer only
```
[the remark this answers is not in the document]
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/save_utils.cpp]: Removed.

**Review by baldersheim — APPROVED** [review]

**Pull request merged**